utilities/expression-helpers.metta

```diff
@@ -113,3 +113,20 @@
     )
   )
 )
+
+;; mutatedGeneCount counts the number of mutated genes and 
+;;        return true only if the number of changed genes is either 0 or 1
+;;        Flattens the two gene lists, original and mutated, so that changes can be compared element by element
+
+(: mutatedGeneCount (-> Expression Expression Bool))
+(= (mutatedGeneCount $original $mutated)

```

Comment:
mutatedGeneCount now doesn't reflect the behavior of the function. Best if you can rename it. And this function is specific for the testing only right? So it's best to keep it just above the test cases you wrote for now. In the near future, when we can put the test cases into their own separate file, this function will be in the test file it self. :)